File: python/beer-song/beer_song.py
def recite(start, take=1):
    lyrics = []

    while take > 0:
        if start == 0:
            lyrics += [
                "No more bottles of beer on the wall, no more bottles of beer.",
                f"Go to the store and buy some more, 99 bottles of beer on the wall."]

            break

        lyrics += [
            f"{bottle_num(start)} {bottle_txt(start)} of beer on the wall, {bottle_num(start)} {bottle_txt(start)} of beer.",
            f"Take {bottle_qty(start - 1)} down and pass it around, {bottle_num(start - 1)} {bottle_txt(start - 1)} of beer on the wall."]

        take -= 1
        start -= 1

        if take > 0:
            lyrics.append("")

    return lyrics


def bottle_num(number):
    return f"{number}" if number > 0 else "no more"


def bottle_txt(number):
    return "bottles" if number != 1 else "bottle"


def bottle_qty(number):
    return "one" if number > 0 else "it"

File: javascript/beer-song/beer-song.js
export const recite = (initialBottlesCount, takeDownCount = 1) => {
  const lyrics = [];

  while (takeDownCount) {
    if (initialBottlesCount === 0) {
      lyrics.push(
        "No more bottles of beer on the wall, no more bottles of beer.",
        "Go to the store and buy some more, 99 bottles of beer on the wall.");

      break;
    }

    const bottles = getBottles(initialBottlesCount);
    lyrics.push(
      `${bottles} of beer on the wall, ${bottles} of beer.`,
      `Take ${initialBottlesCount - 1 ? 'one' : 'it'} down and pass it around, ${getBottles(initialBottlesCount - 1)} of beer on the wall.`);

    takeDownCount--;
    initialBottlesCount--;

    if (takeDownCount > 0)
      lyrics.push("");
  }

  return lyrics;
};

const getBottles = (number) => `${number ? number : 'no more'} ${number != 1 ? 'bottles' : 'bottle'}`;
